server/protocols/webadmin/webadmin-mailbox/src/test/java/org/apache/james/webadmin/service/ExportServiceTest.java

```diff
@@ -0,0 +1,277 @@
+            ZipAssert.assertThatZip(new FileInputStream(fileUrl))
+                .hasNoEntry();
+    }
+
+    @Test
+    void exportUserMailboxesDataShouldProduceAnEmptyZipWhenExistingUserWithoutAnyMailboxes() throws Exception {
+        testee.export(BOB).block();
+
+        String fileUrl = mailetContext.getSentMails().get(0).getMsg().getHeader(CORRESPONDING_FILE_HEADER)[0];
+
+        ZipAssert.assertThatZip(new FileInputStream(fileUrl))
+            .hasNoEntry();
+    }
+
+    @Test
+    void exportUserMailboxesDataShouldProduceAZipWithEntry() throws Exception {
+        ComposedMessageId id = createAMailboxWithAMail(MESSAGE_CONTENT);
+
+        testee.export(BOB).block();
+
+        String fileUrl = mailetContext.getSentMails().get(0).getMsg().getHeader(CORRESPONDING_FILE_HEADER)[0];
+
+        ZipAssert.assertThatZip(new FileInputStream(fileUrl))
+            .containsOnlyEntriesMatching(
+                ZipAssert.EntryChecks.hasName(INBOX + "/").isDirectory(),
+                ZipAssert.EntryChecks.hasName(id.getMessageId().serialize()).hasStringContent(MESSAGE_CONTENT));
+    }
+
+    @Test
+    void exportUserMailboxesDataShouldProduceAFileWithExpectedExtension() throws Exception {
+        createAMailboxWithAMail(MESSAGE_CONTENT);
+
+        testee.export(BOB).block();
+
+        String fileUrl = mailetContext.getSentMails().get(0).getMsg().getHeader(CORRESPONDING_FILE_HEADER)[0];
+
+        assertThat(Files.getFileExtension(fileUrl)).isEqualTo(FileExtension.ZIP.getExtension());
+    }
+
+    @Test
+    void exportUserMailboxesDataShouldProduceAFileWithExpectedName() throws Exception {
+        createAMailboxWithAMail(MESSAGE_CONTENT);
+
+        testee.export(BOB).block();
+
+        String fileUrl = mailetContext.getSentMails().get(0).getMsg().getHeader(CORRESPONDING_FILE_HEADER)[0];
+        File file = new File(fileUrl);
+
+        assertThat(file.getName()).startsWith(FILE_PREFIX + BOB.asString());
+    }
+
+    @Test
+    void exportUserMailboxesWithSizableDataShouldProduceAFile() throws Exception {
+        ComposedMessageId id = createAMailboxWithAMail(TWELVE_MEGABYTES_STRING);
+
+        testee.export(BOB).block();
+        String fileUrl = mailetContext.getSentMails().get(0).getMsg().getHeader(CORRESPONDING_FILE_HEADER)[0];
+
+        ZipAssert.assertThatZip(new FileInputStream(fileUrl))
+            .containsOnlyEntriesMatching(
+                ZipAssert.EntryChecks.hasName(INBOX + "/").isDirectory(),
+                ZipAssert.EntryChecks.hasName(id.getMessageId().serialize()).hasStringContent(TWELVE_MEGABYTES_STRING));
+    }
+
+    @Test
+    void exportUserMailboxesDataShouldDeleteBlobAfterCompletion() throws Exception {
+        createAMailboxWithAMail(MESSAGE_CONTENT);
+
+        testee.export(BOB).block();
+
+        String fileUrl = mailetContext.getSentMails().get(0).getMsg().getHeader(CORRESPONDING_FILE_HEADER)[0];
```

Comment:
is not there a proper API?

Reply:
It doesn't look like so, at least I couldn't find it (if you know please point us to it?)

However, it's not indented well, and actually this could be extracted into a private method, as it's repeated many times in the test. Agreed?